enhancements/quota-management.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Quota Management and Enforcement on Quay
+authors:
+  - "@kwestpha"
+  - "@sleesinc"
+  - "@sdadi"
+reviewers:
+  - "@kwestpha"
+  - "@sleesinc"
+  - "@sdadi"
+approvers:
+  - "@kwestpha"
+  - "@sleesinc"
+  - "@sdadi"
+creation-date: 2021-12-03
+last-updated: 2021-12-03
+status: implementable
+---
+
+# quota-management-and-enforcement
+
+Currently, Quay users have very little control to contain unbounded growth of storage consumption of the registry.
+This proposal is to report storage consumption and contain registry growth based on the configured storage quota limits.  
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+ > 1. Do we have any here?
+
+## Summary
+
+There are no limitations on a Quay organization in terms of their storage consumption. This is a three-phase proposal that
+aims to contain and provide a smart solution to tackle storage related problems for Quay customers.   
+In the first phase i.e, Quota Reporting: a super admin can see the storage consumption of all of their organizations, and a user can see the storage consumption of their organization.
+In the second phase i.e, Quota Management: a superuser can define soft and hard checks for Quay users.
+A soft check implies that the Quay user will be notified if the storage consumption of an organization reaches the configured threshold.
+A hard check implies that the Quay user will not be allowed to push to the registry once the storage consumption reaches the set limit.
+In the third phase, a quay admin can configure auto pruning of the images based on when they were last used to free up quota.
+
+## Motivation
+
+Users can create new organizations and repositories in their home organizations without any practical storage limitation.
+On-prem Quay customers are subject to capacity limits of their environment, specifically backend storage consumption.
+No knowledge about their storage consumption makes it hard for a service owner of a Quay registry to define SLAs and maintain a budget.
+
+### Goals
+
+* A Quay superuser can see the overall storage consumption of the entire registry i.e., an overview of all organizations depicting the aggregate storage consumption of images in an organization.
```

Comment:
A user should also have access to their own resource usage via the API.
So a user should be able to query `/api/v1/repository/<namespace>/<reponame>/usage` or something similar to get the status of his repo.

Reply:
The superuser, or ui view, would do some sort of aggregation on these.

Reply:
Yes, agreed. Will add this in goals.